spec/rubocop/cop/style/redundant_parentheses_spec.rb

```diff
@@ -207,12 +208,35 @@
   it_behaves_like 'redundant', '(~x)', '~x', 'a unary operation'
   it_behaves_like 'redundant', '(-x)', '-x', 'a unary operation'
   it_behaves_like 'redundant', '(+x)', '+x', 'a unary operation'
-  it_behaves_like 'plausible', '(!x.m arg)'
+
+  # No problem removing the parens when it is the only expression
+  it_behaves_like 'redundant', '(!x arg)', '!x arg', 'a unary operation'
+  it_behaves_like 'redundant', '(!x.m arg)', '!x.m arg', 'a unary operation'
+  it_behaves_like 'redundant', '(!super arg)', '!super arg', 'a unary operation'
+  it_behaves_like 'redundant', '(!yield arg)', '!yield arg', 'a unary operation'
+  it_behaves_like 'redundant', '(!defined? arg)', '!defined? arg', 'a unary operation'
+  # Removing the parens leads to semantic differences
+  it_behaves_like 'plausible', '(!x arg) && foo'
+  it_behaves_like 'plausible', '(!x.m arg) && foo'
+  it_behaves_like 'plausible', '(!super arg) && foo'
+  it_behaves_like 'plausible', '(!yield arg) && foo'
+  it_behaves_like 'plausible', '(!defined? arg) && foo'
+  # Removing the parens leads to a syntax error
+  it_behaves_like 'plausible', 'foo && (!x arg)'
+  it_behaves_like 'plausible', 'foo && (!x.m arg)'
+  it_behaves_like 'plausible', 'foo && (!super arg)'
+  it_behaves_like 'plausible', 'foo && (!yield arg)'
+  it_behaves_like 'plausible', 'foo && (!defined? arg)'
```

Comment:
That's just my two cents.

```suggestion
  # No problem removing the parens when it is the only expression.
  it_behaves_like 'redundant', '(!x arg)', '!x arg', 'a unary operation'
  it_behaves_like 'redundant', '(!x.m arg)', '!x.m arg', 'a unary operation'
  it_behaves_like 'redundant', '(!super arg)', '!super arg', 'a unary operation'
  it_behaves_like 'redundant', '(!yield arg)', '!yield arg', 'a unary operation'
  it_behaves_like 'redundant', '(!defined? arg)', '!defined? arg', 'a unary operation'

  # Removing the parens leads to semantic differences.
  it_behaves_like 'plausible', '(!x arg) && foo'
  it_behaves_like 'plausible', '(!x.m arg) && foo'
  it_behaves_like 'plausible', '(!super arg) && foo'
  it_behaves_like 'plausible', '(!yield arg) && foo'
  it_behaves_like 'plausible', '(!defined? arg) && foo'

  # Removing the parens leads to a syntax error.
  it_behaves_like 'plausible', 'foo && (!x arg)'
  it_behaves_like 'plausible', 'foo && (!x.m arg)'
  it_behaves_like 'plausible', 'foo && (!super arg)'
  it_behaves_like 'plausible', 'foo && (!yield arg)'
  it_behaves_like 'plausible', 'foo && (!defined? arg)'
```